lib/samlr/request.rb

```diff
@@ -18,13 +18,17 @@ def body
       @body ||= Samlr::Tools::RequestBuilder.build(options)
     end
 
+    def type
+      "SAMLRequest"
+    end
+
     # Utility method to get the full redirect destination, Request#url("https://idp.example.com/saml", { :RelayState => "https://sp.example.com/saml" })
     def url(root, params = {})
       dest = root.dup
       if dest.include?("?")
```

Comment:
prefer to split data from logic ... 

`"#{dest.include?("?") ? "&" : "?"}#{type}=#{param}`